paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass.cc

```diff
@@ -13,6 +13,56 @@ See the License for the specific language governing permissions and
 limitations under the License. */
 
 #include "paddle/fluid/framework/ir/mkldnn/mkldnn_placement_pass.h"
+#include "paddle/fluid/framework/operator.h"
+
+namespace paddle {
+namespace framework {
+namespace ir {
+
+bool MKLDNNPlacementPass::IsSupport(const Node* op) const {
+  auto op_type = op->Op()->Type();
+
+  auto& all_kernels = OperatorWithKernel::AllOpKernels();
+  auto it = all_kernels.find(op_type);
+  if (it != all_kernels.end()) {
+    for (auto& kernel_pair : it->second) {
+      if (platform::is_cpu_place(kernel_pair.first.place_) &&
+          (kernel_pair.first.library_type_ == LibraryType::kMKLDNN)) {
+        if (op->inputs.size() > 0) {
+          if (op->inputs[0]->IsVar() &&
+              op->inputs[0]->Var()->Name() != "feed" &&
+              kernel_pair.first.data_type_ ==
+                  op->inputs[0]->Var()->GetDataType())
+            return true;
+        } else {
+          return true;
+        }
```

Comment:
Since this is repeated twice in the same function, could you put it into a separate function  with a descriptive name for easier readability?